server.py

```diff
@@ -74,8 +90,14 @@ async def server_loop(self):
         except Exception as err:
             logger.error("Server loop exception occurred:"
                          "{}: {}".format(err.__class__.__name__, err))
+            logger.error("Error details and traceback: {}".format(traceback.format_exc()))
         finally:
-            self.die()
+            if not self._expect_server_loop_death:
```

Comment:
Looks like leftovers from the previous implementation; please remove if not needed

Reply:
I've removed the now-unneeded expect_server_loop_death flag and all references thereto.